src/main/java/com/yanna/stepanova/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.yanna.stepanova.service.impl;
+
+import com.yanna.stepanova.dto.order.CreateOrderRequestDto;
+import com.yanna.stepanova.dto.order.OrderDto;
+import com.yanna.stepanova.dto.order.OrderDtoWithoutOrderItems;
+import com.yanna.stepanova.dto.order.OrderStatusRequestDto;
+import com.yanna.stepanova.mapper.OrderMapper;
+import com.yanna.stepanova.model.CartItem;
+import com.yanna.stepanova.model.Order;
+import com.yanna.stepanova.model.OrderItem;
+import com.yanna.stepanova.model.Status;
+import com.yanna.stepanova.model.User;
+import com.yanna.stepanova.repository.order.OrderRepository;
+import com.yanna.stepanova.service.CartItemService;
+import com.yanna.stepanova.service.OrderItemService;
+import com.yanna.stepanova.service.OrderService;
+import com.yanna.stepanova.service.ShoppingCartService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Transactional
```

Comment:
Apply `@Transactional` only to methods that require transactional behavior, such as those performing create, update, or delete operations
```suggestion
```